### PR TITLE
Student workshop detail

### PR DESCRIPTION
**Pull request by KenPhetmany**
I want to merge my current branch with Admin-workshop, so i'll know the workshop properties i'll be working with

### CODE REVIEW

File: src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	// 				{ title: "Type of Assignment", field: "assignmentType" },
+	// 				{ title: "Group Assignment", field: "isGroupAssignment" },
+	// 				// { title: 'A/NA', field: '' },
+	// 				{ title: "Topics to be discussed", field: "needsHelpWithOptions" },
+	// 				{ title: "Other possible discussions", field: "additionalHelpDetails" }
+	// 			],
+	// 			data: data.map((session: ISessionDetails) => session)
+	// 		});
+	// 	}
+	// }, [data]);
```
Comment (Jahmilli) [inline src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx]: Do we need this here commented out?

File: src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	return (
+		// <div>
+		// 	<Typography variant="h2">Sessions detail</Typography>
+		// 	<EditableTable
+		// 		state={state}
+		// 		setState={setState}
+		// 		options={{ toolbar: false, paging: false }}
+		// 	/>
+			
+		// </div>
```
Comment (Jahmilli) [inline src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx]: Is this needed as well?

File: src/components/containers/StudentDashboard/StudentWorkshopRegistration.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+let workshopData:any;
+
+const StudentWorkshopRegistration: React.FunctionComponent<StudentWorkShopProps> = (props) => {
+    const [sessions, setSessions] = React.useState<Array<ISession>>([]);
+    const context = React.useContext(UserContext);
+
+    React.useEffect(() => {
+        async function callGetWorkshops() {
+            try {
+                workshopData = await getStudentWorkshops(context.userDetails.studentId);
```
Comment (Jahmilli) [inline src/components/containers/StudentDashboard/StudentWorkshopRegistration.tsx]: don't do this. If we need to store state relevant to the component, use a state hook. 
See https://reactjs.org/docs/hooks-state.html
So what you could do (assuming your setState is called setWorkshopData) would be like:
 ```
let workshopData = await getStudentWorkshops(context.userDetails.studentId);
if (workshopData) {
  setWorkshopData(workshopData);
}
```
Also what is the type of workshop Data? You have `any`.

File: src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+type SessionDetails = RouteComponentProps<any> & {};
+
+const icons = {
+	Add: () => <Add /> as React.ReactElement<SvgIconProps>
+};
+
+const StudentWorkshopDetail: React.FunctionComponent<SessionDetails> = (props) => {
+	console.log('state from router is ', props.location.state);
+	const { name, id } = props.location.state;
+	const [state, setState] = React.useState({});
```
Comment (Jahmilli) [inline src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx:31]: what is state being used here for?

File: src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import { withRouter, RouteComponentProps } from 'react-router';
+
+type SessionDetails = RouteComponentProps<any> & {};
+
+const icons = {
+	Add: () => <Add /> as React.ReactElement<SvgIconProps>
+};
+
+const StudentWorkshopDetail: React.FunctionComponent<SessionDetails> = (props) => {
+	console.log('state from router is ', props.location.state);
```
Comment (Jahmilli) [inline src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx]: Best not to keep console logs in unless for debugging purposes.

**Review by Jahmilli — CHANGES_REQUESTED** [review]
good start, theres a few changes, the main one is around setting state which i'd recommend you have a look at.
Also is the StudentWorkshopDetails.tsx finished? lots of commented out stuff there.

**Comment by luke9216** [comment]
Nope fuck off 😉

File: src/components/containers/StudentDashboard/StudentWorkshopRegistration.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import MaterialTable from "material-table";
+import { RouteComponentProps } from "react-router";
+import { Link, withRouter } from "react-router-dom";
+
+type StudentWorkShopProps = RouteComponentProps<any> & {};
+
+let workshopData: any;
+
+const StudentWorkshopRegistration: React.FunctionComponent<StudentWorkShopProps> = props => {
+	const [sessions, setSessions] = React.useState<Array<ISession>>([]);
```
Comment (Jahmilli) [inline src/components/containers/StudentDashboard/StudentWorkshopRegistration.tsx:14]: this isn't being used?? 
Also should it be of type `Array<ISession>` or `Array<IWorkshop>`???

File: src/components/containers/StudentDashboard/StudentWorkshopRegistration.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	const context = React.useContext(UserContext);
+
+	React.useEffect(() => {
+		async function callGetWorkshops() {
+			try {
+				workshopData = await getStudentWorkshops(context.userDetails.studentId);
```
Comment (Jahmilli) [inline src/components/containers/StudentDashboard/StudentWorkshopRegistration.tsx:20]: you should do something like:
```
const workshopData = await getStudentWorkshops(context.userDetails.studentId);
 if (workshopData) {
  setSessions(workshopData);
}

```

File: src/components/pages/StudentHomePage.tsx [hunk 8]
```diff
@@ -6,11 +6,14 @@ import Navbar, { NavbarTab } from '../common/Navbar';
...
 import Footer from '../common/Footer';
 import StudentRegistrationContainer from '../containers/StudentDashboard/StudentRegistrationContainer';
+import StudentWorkshopRegistration from '../containers/StudentDashboard/StudentWorkshopRegistration';
 import navbarTabs from './__data__/data.studentNavbarTabs.json';
 import Auth from '../../logic/functions/core/Auth';
 import LoginPage from './LoginPage';
 import StudentBookingsContainer from '../containers/StudentDashboard/StudentBookingsContainer';
 import StudentDetails from '../presentational/StudentDashboard/StudentDetails';
+import StudentWorkshopDetail from '../presentational/StudentDashboard/StudentWorkshopDetails';
+import { ISessionDetails } from '../../logic/domains/sessionDetails.domain';
```
Comment (Jahmilli) [inline src/components/pages/StudentHomePage.tsx:16]: Where is this being used?

File: src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx (hunk 4 above)
Comment (Jahmilli) [reply src/components/presentational/StudentDashboard/StudentWorkshopDetails.tsx:31]: ??

**Review by Jahmilli — CHANGES_REQUESTED** [review]
it doesnt look like you've made the fixes i requested last week. You shouldnt be setting state in a component by directly assigning a new value. Use the state hook: https://reactjs.org/docs/hooks-state.html

**Comment by KenPhetmany** [comment]
Not needed anymore

**Pull request closed without merging**